astropy/__init__.py

```diff
@@ -172,15 +155,15 @@ def _initialize_astropy():
                 "  pip install -e .\n\nor\n\n"
                 "  python setup.py build_ext --inplace\n\n"
                 "to make sure the extension modules are built "
-            )
-        else:
-            # Outright broken installation, just raise standard error
-            raise
+            ) from None
+
+        # Outright broken installation, just raise standard error
+        raise
 
 
 # Set the bibtex entry to the article referenced in CITATION.
 def _get_bibtex():
-    citation_file = os.path.join(os.path.dirname(__file__), "CITATION")
+    citation_file = pathlib.Path(__file__).parent / "CITATION"
 
     with open(citation_file) as citation:
         refs = citation.read().split("@ARTICLE")[1:]
```

Comment:
It is possible to refactor this more agressively:
```suggestion
    refs = (pathlib.Path(__file__).parent / "CITATION").read_text().split("@ARTICLE")[1:]
```

Reply:
Thanks. That's definitely an improvement.